atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/internalschema/persistence/VersionedInternalSchemaMetadata.java

```diff
@@ -28,11 +31,38 @@
  * Note that additive changes may be made to {@link com.palantir.atlasdb.internalschema.InternalSchemaMetadata}
  * without requiring an increase in the version number; increases are only needed when wire-compatibility would be
  * broken, or it is not possible to assign a sensible default value for events occurring in the past.
+ *
+ * Note: Please be careful when using this object's equals() method, as that compares the exact bytes written
+ * in the payload, and thus might consider semantically equivalent payloads to be different.
  */
 @Value.Immutable
 @JsonSerialize(as = ImmutableVersionedInternalSchemaMetadata.class)
 @JsonDeserialize(as = ImmutableVersionedInternalSchemaMetadata.class)
 public interface VersionedInternalSchemaMetadata {
     int version();
     byte[] payload();
+
+    /**
+     * Returns the underlying {@link InternalSchemaMetadata} for this {@link VersionedInternalSchemaMetadata},
+     * provided we are able to deserialize it. Otherwise returns empty.
+     */
+    @Value.Lazy
+    default Optional<InternalSchemaMetadata> underlyingData() {
```

Comment:
I think you want a `@JsonIgnore` on here.

Reply:
Sure, will add this.